autogpt_platform/frontend/src/lib/autogpt-server-api/baseClient.ts

```diff
@@ -238,6 +240,31 @@ export default class BaseAutoGPTServerAPI {
     return this._request("GET", path, query);
   }
 
+  // Scheduling request
+  async createSchedule(
+    graphId: string,
+    schedule: ScheduleCreatable,
+  ): Promise<{ id: string }> {
+    return this._request(
+      "POST",
+      `/graphs/${graphId}/schedules?cron=${schedule.cron}`,
+      {
+        input_data: schedule.input_data,
+      },
+    );
+  }
+
+  async updateSchedule(
+    scheduleId: string,
+    update: ScheduleUpdateable,
+  ): Promise<{ id: string }> {
+    return this._request("PUT", `/graphs/schedules/${scheduleId}`, update);
+  }
+
+  async getSchedules(graphId: string): Promise<{ [key: string]: string }> {
```

Comment:
After seeing it used by the client, I think it makes much more sense to make the API route simply `/schedules` without the graph intervention, I can add `graph_id` as a query parameter for GET request as an optional filtering clause.